File: src/main/java/com/ByTech/ByTech/Cart/models/ShoppingCart.java
package com.ByTech.ByTech.Cart.models;

import com.ByTech.ByTech.User.models.UserModel;
import jakarta.persistence.*;
import org.antlr.v4.runtime.misc.NotNull;
import org.hibernate.annotations.GenericGenerator;

@Entity
public class ShoppingCart {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false, cascade = CascadeType.DETACH, fetch = FetchType.EAGER)
    @JoinColumn(name = "product_id", referencedColumnName = "id") // Aquí es donde se establece la relación correcta
    private Product product;

    @ManyToOne(optional = false, cascade = CascadeType.DETACH, fetch = FetchType.EAGER)
    @JoinColumn(name = "client_id", referencedColumnName = "id")
    private UserModel client;

    @Column
    private int amount;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Product getProduct() {
        return product;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public UserModel getClient() {
        return client;
    }

    public void setClient(UserModel client) {
        this.client = client;
    }

    public int getAmount() {
        return amount;
    }

    public void setAmount(int amount) {
        this.amount = amount;
    }
}

File: Movil/backend/ByTech_movil/src/main/kotlin/com/example/ByTech_movil/ShoppingCart/models/ShoppingCart.kt
package com.example.ByTech_movil.ShoppingCart.models

import com.example.ByTech_movil.Product.models.Product
import com.example.ByTech_movil.User.models.UserModel
import jakarta.persistence.*

@Entity
class ShoppingCart {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @ManyToOne(optional = false, cascade = [CascadeType.DETACH], fetch = FetchType.EAGER)
    @JoinColumn(
        name = "product_id",
        referencedColumnName = "id"
    ) // Aquí es donde se establece la relación correcta
    var product: Product? = null

    @ManyToOne(optional = false, cascade = [CascadeType.DETACH], fetch = FetchType.EAGER)
    @JoinColumn(name = "client_id", referencedColumnName = "id")
    var client: UserModel? = null

    @Column
    var amount: Int = 0
}